pkg/lightning/config/config.go

```diff
@@ -200,12 +207,6 @@ func (d *DBStore) adjust(
 	case "":
 	case "false":
 		d.Security.TLSConfig = nil
-		d.Security.CAPath = ""
```

Comment:
We should keep them. Because the real effective object is `TLSConfig`, and it's built from these fields in https://github.com/pingcap/tidb/blob/761b5926a73d15ef165f525fdada5a0f7fa3e221/lightning/pkg/server/lightning.go#L394 and https://github.com/pingcap/tidb/blob/761b5926a73d15ef165f525fdada5a0f7fa3e221/lightning/pkg/server/lightning.go#L503, while this `adjust` is called at https://github.com/pingcap/tidb/blob/761b5926a73d15ef165f525fdada5a0f7fa3e221/lightning/pkg/server/lightning.go#L355 which is before `run`. So only assigning `Security.TLSConfig` to `nil` is not enough, because `BuildTLSConfig` will re-create the `TLSConfig`

Reply:
@lance6716 Thanks for the review.
I was mistaken. I should have discussed this issue before I started working on it.
I have two questions.
Is TLS always enabled for connections to PD?
Do we need to add a CAPath property dedicated to connections to TiDB to solve this problem?

Reply:
> Is TLS always enabled for connections to PD?

it's controlled by `[security]` section in configuration file

> Do we need to add a CAPath property dedicated to connections to TiDB to solve this problem?

We already have it, it's `[tidb.security]`. And when `[tidb.security]` is not set, the fallback value is `[security]`. The original issue is related to `tidb.tls` configuration which will modify the behaviour of `[tidb.security]`. But when fallback happens, we don't want `tidb.tls` affects `[security]`.

https://docs.pingcap.com/tidb/stable/tidb-lightning-configuration